test/ir_Haier_test.cpp

```diff
@@ -1223,7 +1231,148 @@ TEST(TestUtils, Housekeeping) {
   ASSERT_EQ("HAIER_AC176", typeToString(decode_type_t::HAIER_AC176));
   ASSERT_EQ(decode_type_t::HAIER_AC176, strToDecodeType("HAIER_AC176"));
   ASSERT_TRUE(hasACState(decode_type_t::HAIER_AC176));
-  ASSERT_FALSE(IRac::isProtocolSupported(decode_type_t::HAIER_AC176));
+  ASSERT_TRUE(IRac::isProtocolSupported(decode_type_t::HAIER_AC176));
   ASSERT_EQ(kHaierAC176Bits, IRsend::defaultBits(decode_type_t::HAIER_AC176));
   ASSERT_EQ(kNoRepeat, IRsend::minRepeats(decode_type_t::HAIER_AC176));
 }
+
+TEST(TestHaierAC176Class, BuildKnownState) {
+  IRHaierAC176 ac(kGpioUnused);
+  // Ref: https://github.com/crankyoldgit/IRremoteESP8266/issues/1480#issuecomment-884920033
+  // heat, 24 C, fan max
+  ac.setTemp(24);
+  ac.setMode(kHaierAcYrw02Heat);
+  ac.setFan(kHaierAcYrw02FanHigh);
+  EXPECT_TRUE(ac.validChecksum(ac.getRaw()));
+  EXPECT_EQ(
+      "Power: On, Button: 4 (Fan), Mode: 4 (Heat), Temp: 24C, Fan: 1 (High), "
+      "Turbo: 0 (Off), Swing: 0 (Off), Sleep: Off, Health: On, "
+      "Timer Mode: 0 (N/A), On Timer: Off, Off Timer: Off",
+      ac.toString());
+  /* Disabled pending:
+     https://github.com/crankyoldgit/IRremoteESP8266/issues/1480#issuecomment-885636790
```

Comment:
No resolution here?

Reply:
Sadly no. :-(

Possible case of poorly described data being reported as the set/reported Temperature doesn't match the `state[]` data supplied.

Also, no simple/easy other `state[]`s to copy from to try to rebuild a known good state from.